TAGradingServer/account/submit/admin-gradeable.php

```diff
@@ -0,0 +1,411 @@
+<?php
+
+// TODO MORE error checking
+// TODO Make sure transactions are in the right spots
+// TODO functionalize more
+
+include "../../toolbox/functions.php";
+
+check_administrator();
+
+if (!isset($_POST['csrf_token']) || $_POST['csrf_token'] !== $_SESSION['csrf']) {
+    die("invalid csrf token");
+}
+
+ class Gradeable{
+     protected $g_id;
+     protected $g_title;
+     protected $g_overall_ta_instr;
+     protected $g_use_teams;
+     protected $g_gradeable_type; 
+     protected $g_min_grading_group;
+     protected $g_grade_by_registration;
+     protected $g_grade_start_date;
+     protected $g_grade_released_date;
+     protected $g_syllabus_bucket;
+     
+    function __construct(& $params){
```

Comment:
Don't pass by reference in PHP unless you're actually modifying the paramter (which you aren't). It'll incur a (slight) performance penalty as opposed to if you had just done pass by value since there's no modification.